### PR TITLE
Fix/227

### PR DESCRIPTION
**Pull request by xross**
Proposed fix to issue #227.

Also includes directed test.

(Also some whitespace fixes)

Passes pytest --smoke

### CODE REVIEW

File: lib_xud/src/core/included/XUD_Token_Setup_DI.S [hunk 1]
```diff
@@ -24,24 +24,25 @@ ClearStall:                                     // CRC OK
...
+    stw        r1, r11[r10]
 
 SendSetupAck:
     ldc        r11, USB_PIDn_ACK
     outpw      res[TXD], r11, 8
 
 StoreTailData:                                  // TODO: don't assume setups are 8 bytes + crc
-    ldc        r11, 0
-    stw        r11, r5[r10]                     // Clear ready
+    stw        r1, r5[r10]                      // Clear ready
```
Comment (mbanth) [inline lib_xud/src/core/included/XUD_Token_Setup_DI.S:38]: Does anything branch to `StoreTailData`?  I do not see a branch to it in this file, but i know that some pieces of code are `#include`d by others.  If a branch to `StoreTailData` does exist, is there a guarantee that r1 has zero in it?

**Review by mbanth — APPROVED** [review]
One question which is speculative and included only as a precaution.

**Review by DaveAtkinsonXMOS — APPROVED** [review]
Looks good to me. I've tested: xcore.ai and xcore200, on Linux (x86_64 & arm) and windows, using AN00124 CDC VCOM class and AN00129 HID class, and they all work.

File: lib_xud/src/core/included/XUD_Token_Setup_DI.S (hunk 1 above)
Comment (xross) [reply lib_xud/src/core/included/XUD_Token_Setup_DI.S:38]: There are no branches to it but the code falls to it from above.
I tend to use a fair amount of labels when I'm coding asm for my sanity, readability and also debugging - easier to see when the code is when breaking.

Comment (mbanth) [reply lib_xud/src/core/included/XUD_Token_Setup_DI.S:38]: I wondered if that was the case.  No worries then.

**Pull request merged**